src/netspresso_trainer/losses/common.py

```diff
@@ -17,3 +17,35 @@ def forward(self, out: Dict, target: torch.Tensor) -> torch.Tensor:
         pred = out['pred']
         loss = self.loss_fn(pred, target)
         return loss
+
+
+class SigmoidFocalLoss(nn.Module):
+    def __init__(self, alpha, gamma):
+        super(SigmoidFocalLoss, self).__init__()
+        self.alpha = alpha
+        self.gamma = gamma
+
+    def forward(self, out: Dict, target: torch.Tensor, reduction='mean'):
+        pred = out['pred']
+        assert pred.shape == target.shape, 'Tensor shapes of prediction and target must be same for SigmoidFocalLoss.'
+
+        p = torch.sigmoid(pred)
+        ce_loss = F.binary_cross_entropy_with_logits(pred, target, reduction="none")
+        p_t = p * target + (1 - p) * (1 - target)
+        loss = ce_loss * ((1 - p_t) ** self.gamma)
+
+        if self.alpha >= 0:
+            alpha_t = self.alpha * target + (1 - self.alpha) * (1 - target)
+            loss = alpha_t * loss
+
+        if reduction == 'none':
```

Comment:
Why is here with string none ??

Reply:
I referenced [this code](https://github.com/pytorch/vision/blob/main/torchvision/ops/focal_loss.py), 
it because`reduction` is `str` type.